console/console_test.go

```diff
@@ -294,3 +294,49 @@ func TestPrettyError(t *testing.T) {
 		t.Fatalf("pretty error mismatch: have %s, want %s", output, want)
 	}
 }
+
+// Tests that tests if the number of indents for JS input is calculated correct.
+func TestIndenting(t *testing.T) {
+	testCases := []struct {
+		input               string
+		expectedIndentCount int
+	}{
+		{`var a = 1;`, 0},
+		{`"some string"`, 0},
+		{`"some string with (parentesis`, 0},
+		{`"some string with newline
+		("`, 0},
+		{`function v(a,b) {}`, 0},
+		{`function f(a,b) { var str = "asd("; };`, 0},
+		{`function f(a) {`, 1},
+		{`function f(a, function(b) {`, 2},
+		{`function f(a, function(b) {
+		     var str = "a)}";
+		  });`, 0},
+		{`function f(a,b) {
+		   var str = "a{b(" + a, ", " + b;
+		   }`, 0},
+		{`var str = "\"{"`, 0},
+		{`var str = "'("`, 0},
+		{`var str = "\\{"`, 0},
+		{`var str = "\\\\{"`, 0},
+		{`var str = 'a"{`, 0},
+		{`var obj = {`, 1},
+		{`var obj = { {a:1`, 2},
+		{`var obj = { {a:1}`, 1},
+		{`var obj = { {a:1}, b:2}`, 0},
+		{`var obj = {}`, 0},
+		{`var obj = {
+			a: 1, b: 2
+		}`, 0},
+		{`var test = }`, -1},
+		{`var str = "a\""; var obj = {`, 1},
+	}
+
+	for i, tc := range testCases {
+		counted := countIndents(tc.input)
+		if counted != tc.expectedIndentCount {
+			t.Fatalf("invalid indenting for case %d, have %d, want %d", i, counted, tc.expectedIndentCount)
```

Comment:
Just two tiny nitpicks that diverge from the stdlib test styles:

 * Usually `tt` is used as the test case being tested. (i.e. for i, tt := range testCases)
 * When printing errors, it's usually formatted as `"test %d: invalid indenting: have %d, want %d"`
   * This way if you use Fatalf but ErrorF, then multiple failures can be matched up easily without the test number being at different positions based on the error text.

Ah, also please use `t.Errorf` here. That causes the test to fail but nonetheless runs all the test cases (Fatal aborts immediately). This is usually useful to see multiple issues at once and not have to run&fix one by one.